code/game/atoms_movable.dm

```diff
@@ -662,3 +662,212 @@
+
+
+		else if(isobj(target))  // don't crush things on the floor, that'd probably be annoying
+			var/obj/O = target
+			O.take_damage(damage_to_deal, BRUTE, "", FALSE)
+		else
+			continue
+
+		target.visible_message(
+			"<span class='danger'>[target] is crushed by [src]!</span>",
+			"<span class='userdanger'>[src] crushes you!</span>",
+			"<span class='warning'>You hear a loud crunch!</span>"
+		)
+
+	tilt_over(target_turf, angle, should_rotate, rightable, block_interactions_until_righted)
+	// for things that trigger on Crossed()
+	if(!has_tried_to_move)
+		Move(target_turf, crush_dir)
+
+	return TRUE
+
+#undef NO_CRUSH_DIR
+
+/**
+ * Tip over an atom without too much fuss. This won't cause damage to anything, and just rotates the thing and (optionally) adds the component.
+ *
+ * Arguments:
+ * * target - The turf to tilt over onto
+ * * rotation_angle - The angle to rotate by. If not given, defaults to random rotating by 90 degrees clockwise or counterclockwise
+ * * should_rotate - Whether or not we should rotate at all
+ * * rightable - Whether or not this object should be rightable, attaching the tilted component to it
+ * * block_interactions_until_righted - If true, this object will need to be righted before it can be interacted with
+ */
+/atom/movable/proc/tilt_over(turf/target, rotation_angle, should_rotate, rightable, block_interactions_until_righted)
+	visible_message("<span class='danger'>[src] tips over!</span>", "<span class='danger'>You hear a loud crash!</span>")
+	playsound(src, "sound/effects/bang.ogg", 100, TRUE)
+	var/rot_angle = rotation_angle ? rotation_angle : pick(90, -90)
+	if(should_rotate)
+		var/matrix/to_turn = turn(transform, rot_angle)
+		animate(src, transform = to_turn, 0.2 SECONDS)
+	if(target && target != get_turf(src))
+		throw_at(target, 1, 1, spin = FALSE)
+	if(rightable)
+		layer = ABOVE_MOB_LAYER
+		AddComponent(/datum/component/tilted, 14 SECONDS, block_interactions_until_righted, rot_angle)
+
+/// Untilt a tilted object.
+/atom/movable/proc/untilt(mob/living/user, duration = 10 SECONDS)
+	if(!GetComponent(/datum/component/tilted))
+		return
+	if(SEND_SIGNAL(src, COMSIG_MOVABLE_TRY_UNTILT, user) & COMPONENT_BLOCK_UNTILT)
+		return
+	if(!istype(user) || !Adjacent(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+		return
+
+	if(user)
+		user.visible_message(
+			"[user] begins to right [src].",
+			"You begin to right [src]."
+		)
+		if(!do_after(user, duration, TRUE, src))
+			return
+		user.visible_message(
+			"<span class='notice'>[user] rights [src].</span>",
+			"<span class='notice'>You right [src].</span>",
+			"<span class='notice'>You hear a loud clang.</span>"
+		)
+
+	unbuckle_all_mobs(TRUE)
+
+	SEND_SIGNAL(src, COMSIG_MOVABLE_UNTILTED, user)
+
+	layer = initial(layer)
+
+	var/matrix/M = matrix()
+	M.Turn(0)
+	transform = M
+
+/// useful callback for things that want special behavior on crush
+/atom/movable/proc/on_crush_thing(atom/thing)
+	return
+
+/obj/item/tilter
```

Comment:
I don't think these objects and procs below are used.

Reply:
Yes this this a debug item and probably useless now that the build mode tools are in. I'll remove these when I can.